Add trademark note for Build Scan in documentation (#18433)
gradle/gradle: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/viewing_debugging_dependencies.adoc

Gradle provides sufficient tooling to navigate large dependency graphs and mitigate situations that can lead to link:https://en.wikipedia.org/wiki/Dependency_hell[dependency hell].
Users can choose to render the full graph of dependencies as well as identify the selection reason and origin for a dependency.
The origin of a dependency can be a declared dependency in the build script or a transitive dependency in graph plus their corresponding configuration.

> Gradle offers both capabilities through visual representation via Build Scan™ and as command line tooling.

Reviewer: ```suggestion
Gradle offers both capabilities through visual representation with command-line tooling and the Build Scan™ service.
```
